lib/surgex/parser/parsers/contain_parser.ex

```diff
@@ -0,0 +1,15 @@
+defmodule Surgex.Parser.ContainParser do
+  @moduledoc """
+  Checks if the given parameter's value is on the list of allowed values
+  """
+
+  def call(nil, _allowed_values), do: {:error, :value_is_not_allowed}
+  def call(_, []), do: {:error, :value_is_not_allowed}
```

Comment:
This is actually an argument error and it should throw on compilation. You may remove this case and add `and length(allowed_values) > 0` to `when` in clause below.